lib/rules/no-typos.js

```diff
@@ -54,6 +74,14 @@ module.exports = {
           const propertyName = node.property.name;
           reportErrorIfCasingTypo(node, propertyName);
         }
+      },
+
+      MethodDefinition: function (node) {
+        if (!utils.isES6Component(node.parent.parent) || utils.isReturningJSX(node.parent.parent)) {
```

Comment:
Do we need the check for `utils.isReturningJSX()` ? This is used to detect Stateless Function Component, which does not contain user defined lifecycle methods.